tests/testpathgradientbrush.c

```diff
@@ -0,0 +1,1685 @@
+    assert (status == InvalidParameter);
+
+    status = GdipCreatePathGradientFromPath (path, NULL);
+    assert (status == InvalidParameter);
+
+    GdipDeleteBrush ((GpBrush *)brush);
+    GdipDeletePath (path);
+    GdipDeletePath (emptyPath);
+}
+
+static void test_getPathGradientCenterColor ()
+{
+    GpStatus status;
+    GpPathGradient *brush;
+    ARGB centerColor;
+
+    GdipCreatePathGradient (threePoints, 3, WrapModeTileFlipX, &brush);
+
+    status = GdipGetPathGradientCenterColor (NULL, &centerColor);
+    assert (status == InvalidParameter);
+
+    status = GdipGetPathGradientCenterColor (brush, NULL);
+    assert (status == InvalidParameter);
+
+    GdipDeleteBrush ((GpBrush *) brush);
+}
+
+static void test_setPathGradientCenterColor ()
+{
+    GpStatus status;
+    GpPathGradient *brush;
+    ARGB centerColor;
+
+    GdipCreatePathGradient (threePoints, 3, WrapModeTileFlipX, &brush);
+
+    status = GdipSetPathGradientCenterColor (brush, 2);
+    assert (status == Ok);
+
+    GdipGetPathGradientCenterColor (brush, &centerColor);
+    assert (centerColor == 2);
+
+    status = GdipSetPathGradientCenterColor (NULL, 1);
+    assert (status == InvalidParameter);
+
+    GdipDeleteBrush ((GpBrush *) brush);
+}
+
+static void test_getPathGradientSurroundColorCount ()
+{
+    GpStatus status;
+    GpPathGradient *brush;
+    INT count;
+
+    // Three points.
+    GdipCreatePathGradient (threePoints, 3, WrapModeTileFlipX, &brush);
+
+    status = GdipGetPathGradientSurroundColorCount (brush, &count);
+    assert (status == Ok);
+    assert (count == 3);
+    GdipDeleteBrush ((GpBrush *) brush);
+
+    // Two points.
+    GdipCreatePathGradient (threePoints, 2, WrapModeTileFlipXY, &brush);
+
+    status = GdipGetPathGradientSurroundColorCount (brush, &count);
+    assert (status == Ok);
+    assert (count == 2);
+
+    // Negative tests
+    status = GdipGetPathGradientSurroundColorCount (NULL, &count);
+    assert (status == InvalidParameter);
+
+    status = GdipGetPathGradientSurroundColorCount (brush, NULL);
+    assert (status == InvalidParameter);
+
+    GdipDeleteBrush ((GpBrush *) brush);
+}
+
+static void test_getPathGradientSurroundColorsWithCount ()
+{
+    GpStatus status;
+    GpPathGradient *brush;
+    ARGB colors[3] = {0, 0, 0};
+    INT count = 3;
+
+    // Three points.
+    GdipCreatePathGradient (threePoints, 3, WrapModeTileFlipX, &brush);
+
+    status = GdipGetPathGradientSurroundColorsWithCount (brush, colors, &count);
+    assert (status == Ok);
+    assert (count == 1);
+    assert (colors[0] == 0xffffffff);
+    assert (colors[1] == 0xffffffff);
+    assert (colors[2] == 0xffffffff);
+
+    // Negative tests
+    status = GdipGetPathGradientSurroundColorsWithCount (NULL, colors, &count);
+    assert (status == InvalidParameter);
+
+    status = GdipGetPathGradientSurroundColorsWithCount (brush, NULL, &count);
+    assert (status == InvalidParameter);
+
+    // FIXME: this causes GDI+ to crash.
```

Comment:
/cc @mellinoe also a bug in GDI+

Do we want to simulate a crash in libgdiplus as well?